spring-integration-core/src/main/java/org/springframework/integration/json/JsonPropertyAccessor.java

```diff
@@ -117,6 +120,22 @@ private static Integer maybeIndex(String name) {
 		}
 	}
 
+	/**
+	 * Check if the string is a numeric representation (all digits) or not.
+	 */
+	private static boolean isNumeric(String str) {
+		if (str == null || str.isEmpty()) {
```

Comment:
This can be replaced with `org.springframework.util.StringUtils.hasText()`

Reply:
Good point. But I think using `org.springframework.util.StringUtils.hasLength()` will be even better. Because `hasText()` will check for non empty string and additionally iterate over it looking for non-whitespace characters. Which is a waste of CPU in cases when we get all-whitespace strings here like "_____".
Replaced with `hasLength()`.